crates/ra_hir/src/ty.rs

```diff
@@ -1588,9 +1588,17 @@ impl<'a, D: HirDatabase> InferenceContext<'a, D> {
                             _ => Ty::Unknown,
                         }
                     }
-                    UnaryOp::Not if inner_ty == Ty::Bool => Ty::Bool,
-                    // TODO: resolve ops::Not trait for inner_ty
-                    UnaryOp::Not => Ty::Unknown,
+                    UnaryOp::Not => {
+                        match inner_ty {
+                            Ty::Bool
+                            | Ty::Int(primitive::UncertainIntTy::Unknown)
```

Comment:
These I think could be `Ty::Int(_)`?

Reply:
Yep, you're absolutely right. Off to fixing that now